cider/tests/CiderDateFunctionTest.cpp

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <cstddef>
+#include <string>
+#include "function/datetime/DateAdd.h"
+#include "type/data/sqltypes.h"
+#include "type/plan/Analyzer.h"
+#define CIDERBATCH_WITH_ARROW
+
+#include <gtest/gtest.h>
+#include "CiderAggTestHelper.h"
+#include "TestHelpers.h"
+#include "cider/CiderTypes.h"
+
+#ifndef BASE_PATH
+#define BASE_PATH "./tmp"
+#endif
+
+extern bool g_is_test_env;
+extern bool g_enable_watchdog;
+extern size_t g_big_group_threshold;
+extern size_t g_watchdog_baseline_max_groups;
+
+using namespace TestHelpers;
+
+size_t ELEMENT_NUM = 5;
+
+static const std::shared_ptr<CiderAllocator> allocator =
+    std::make_shared<CiderDefaultAllocator>();
+
+class CiderDateFunctionTest : public ::testing::Test {
+ protected:
+  static MockTable* MockTableForTest;
+
+  static void SetUpTestSuite() {
+    if (nullptr == MockTableForTest) {
+      std::vector<std::string> col_names = {"date_notnull", "date_null"};
+
+      std::vector<SQLTypeInfo> col_types = {
+          SQLTypeInfo(kBIGINT, true),
+          SQLTypeInfo(kBIGINT, false),
+      };
+
+      int64_t* bigint_ptr = new int64_t[ELEMENT_NUM];
+
+      for (size_t i = 0; i < ELEMENT_NUM; ++i) {
+        bigint_ptr[i] = i * 86400 + i * 3600 + i * 60 + i;
+      }
+
```

Comment:
Test for cider date type? I think it's not valid date type value.

Reply:
date extr include time field, so construct hour/minute/second